libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Skills/AllowedCallersClaimsValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+
+namespace Microsoft.Bot.Builder.Integration.Runtime.Skills
+{
+    /// <summary>
+    /// Claims validator that adds application level authorization based on a simple list
+    /// of application Ids that are allowed to call. 
+    /// </summary>
+    internal class AllowedCallersClaimsValidator : ClaimsValidator
```

Comment:
Why isn't this code added to the Microsoft.Bot.Connector?

Reply:
because we are not changing the GA product until we preview the product and get feedback. There are many classes whose final location will be in other places but we kept everything in the preview assemblies to bake them. We discussed this.

Reply:
tl;dr: wontfix til R13. Agree with the sentiment.